packages/expo-cli/src/commands/eas-build/build/templates/gradleContent.ts

```diff
@@ -17,18 +17,27 @@ android {
   }
 }
 
-project.afterEvaluate {
-  android.signingConfigs.release { config ->
-    def debug = gradle.startParameter.taskNames.any { it.toLowerCase().contains('debug') }
+def isEasBuildConfigured = false
 
-    if (debug) {
-      return
-    }
+tasks.whenTaskAdded {
```

Comment:
Did you identified how react-native-config affected our script and what was the actual reason for signing with debug keystore?

- Did it stop script not to execute at all?
- Did it override values set by our script
- Did it assign release keystore (to debug one) before our script had a chance to do that
- sth else?

Reply:
Like I mentioned in the issue, this is the minimum code which triggers the issues:

```gradle
tasks​.​whenTaskAdded​ { task ​->​
    ​//​ Nothing here​
}
```

Like delete everything in it's Gradle config, place this much and the app is signed with debug key. That doesn't leave me with much to go on.

All of our code is still executed after the tasks are added. I'm afraid that without deep knowledge of how the Gradle system works, I don't have an answer to those questions.